packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -91,16 +105,45 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
+        const { paymentData } = payment;
+        const isVaultedFlow =
+            paymentData && isVaultedInstrument(paymentData) && isHostedInstrumentLike(paymentData);
+
         if (!this.orderId) {
-            throw new PaymentMethodInvalidError();
+            if (isVaultedFlow) {
+                this.orderId = await this.createOrder();
+            } else {
+                throw new PaymentMethodInvalidError();
+            }
+        }
+
+        let isTrustedVaultingFlow = true;
+
+        if (paymentData && isVaultedInstrument(paymentData)) {
+            const instruments = state.getInstruments();
+
+            const { trustedShippingAddress } =
+                instruments?.find(({ bigpayToken }) => bigpayToken === paymentData.instrumentId) ||
+                {};
+
+            isTrustedVaultingFlow = Boolean(trustedShippingAddress);
         }
 
         try {
+            const paymentPayload =
+                paymentData &&
+                isVaultedInstrument(paymentData) &&
+                isHostedInstrumentLike(paymentData) &&
+                isTrustedVaultingFlow
```

Comment:
```suggestion
                isVaultedFlow && isTrustedVaultingFlow
```

Reply:
It does not work as we want. Typescript does not work in this way